syntax = "proto3";

//python -m grpc_tools.protoc -I. --python_out=. --grpc_python_out=. task.proto

service TaskService{

//  test
  rpc task_test(TaskRequest)returns(CommonReply){}
  rpc task_transfer_file(File)returns(CommonReply){}
  rpc send_image(Image)returns(CommonReply){}
  rpc task_yolox_vedio(stream Image)returns(stream Image){}

//  task
  rpc task_yolox_image(Image)returns(Image){}
  rpc task_lic_detect(Image)returns(Image){}
  rpc task_linear_regression(Null)returns(CommonReply){}
  rpc task_num_detect(Null)returns(CommonReply){}
  rpc task_monet_transfer(Null)returns(CommonReply){}
  rpc task_yolo5(File)returns(Image){}
  rpc task_compose(Image_x2)returns(Image){}
  rpc task_style_transfer(File_x2)returns(Image){}

//  face_reco
  rpc task_face_recognition(FaceRecoRequest)returns(FaceRecoReply){}

//  keep_alive
  rpc keep_alive(HeartBeat)returns(CommonReply){}
  rpc update_tasks(TaskPackage)returns(CommonReply){}

}

message TaskPackage{
  bool add_task = 1; // 添加任务为True,删减任务为False
  string tasks = 2; // 任务编号 用‘，’隔开 例 ‘1,3,4’
  Address addr = 3;
}

message HeartBeat{
  string name = 1;
  Address addr = 2;
  Resource res = 3;
  string tasks = 4; //当前节点还需要处理的任务
}

// 人脸识别请求
message FaceRecoRequest{
  bytes img = 1;
  string names = 2;
  int32 frame_cnt = 3;
  string node_name = 4;
}

//人脸识别响应
message FaceRecoReply{
  bytes img = 1;
  bool success = 2;
}

message File_x2{
  File content=1;
  File style=2;
}

message Image_x2{
  bytes img = 1;
  bytes img_compose = 2;
}

message Null{

}

message Image{
  bytes img = 1;
}

message TaskRequest{
  int32 task_id = 1;
  string task_name = 2;
}

message CommonReply{
  bool success = 1;
}

message File{
  //
  string file_name = 1;
  bytes file_data = 2;
}


message Address{
  string ip = 1;
  int32 port = 2;
}

// 所有硬件资源
message Resource{
  CPU cpu = 1;
  Memory mem = 2;
  Disc disc = 3;
}


message CPU{
  float use_ratio = 1;  //使用率
  int32 real_num = 2;   //物理个数
  int32 logic_num = 3;  //逻辑个数
}

message Memory{
  int64 total = 1;
  int64 available = 2;
}

message Disc{
  int64 total = 1;
  int64 available = 2;
}